pkg/scheduler/apis/config/v1beta1/defaults.go

```diff
@@ -158,6 +158,15 @@ func SetDefaults_KubeSchedulerConfiguration(obj *v1beta1.KubeSchedulerConfigurat
 	}
 }
 
+func SetDefaults_DefaultPreemptionArgs(obj *v1beta1.DefaultPreemptionArgs) {
+	if obj.MinCandidateNodesPercentage == nil {
+		obj.MinCandidateNodesPercentage = pointer.Int32Ptr(10)
+	}
+	if obj.MinCandidateNodesAbsolute == nil {
+		obj.MinCandidateNodesAbsolute = pointer.Int32Ptr(100)
+	}
+}
+
```

Comment:
do these match the current behavior or do they represent a change in scheduler behavior?

Reply:
This is a change in the scheduler behavior, but it is a change to a scoring behavior which is by definition best effort.

Reply:
Let me expand on this, the invariants that are documented with regard to preemption are two (https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/): 1) preempt lower priority pods, 2) respect PDB whenever possible. Those two are preserved. Beyond that, selecting which exact pods to preempt isn't part of the contract, it is based on a heuristic.

Reply:
I see, so this is not saying we will only search 10% / 100 nodes to find candidate nodes, it is saying once we have 10% / 100 of nodes as candidates, we'll stop looking

Reply:
yes, exactly.

Reply:
It is also worth noting that we'll potentially keep going even after 10% / 100 nodes if we don't have a PDB-respecting candidate. So the 10% / 100 is only a lower limit (the upper limit being the entire cluster, which is the preemption behaviour in-tree).